### PR TITLE
EZAF-1822: Decouple tenant services

### PR DESCRIPTION
**Pull request by apurwaj2**
Adding namespace configuration to tenant operator and tenant validator.

### CODE REVIEW

File: charts/tenant-operator/README.md [hunk 1]
```diff
@@ -3,14 +3,12 @@
...
 
-#### NOTE: This will create the helm chart in the `default` namespace.
 The chart installation will create Tenant operator and tenant validator webhook.
-All the resources defined in the chart are created in `hpe-system` namespace.
-
+All the resources defined in the chart are created in <namespace> namespace.
 
 ## Uninstalling the Chart
-`helm delete tenant-operator`
+`helm delete tenant-operator -n <namespace>`
```
Comment (idzikovskyi) [inline charts/tenant-operator/README.md]: I'd say this piece needs to be wrapped in three backticks to be block of code, rather than inline piece.

**Review by idzikovskyi — APPROVED** [review]
In general it looks good, but I'd ask @AlexNavara to also give feedback on this, since he's more familiar with the problem of passing namespaces this way in our projects.

File: charts/tenant-operator/README.md (hunk 1 above)
Comment (apurwaj2) [reply charts/tenant-operator/README.md]: Thanks, Ivan!

File: charts/tenant-operator/templates/_helpers.tpl [hunk 2]
```diff
@@ -46,7 +46,7 @@ return name for validator
 return namespace where the tenant operator and validator must be deployed
 */}}
 {{- define "tenant-operator.namespace" -}}
```
Comment (AlexNavara) [inline charts/tenant-operator/templates/_helpers.tpl]: Do we need this function anymore?

File: charts/tenant-operator/templates/operator_rbac.yaml [hunk 3]
```diff
@@ -95,7 +95,7 @@ metadata:
 subjects:
   - kind: ServiceAccount
     name: {{ include "tenant-operator.hpePrefix" . }}
-    namespace: hpe-system
+    namespace: {{ include "tenant-operator.namespace" . }}
```
Comment (AlexNavara) [inline charts/tenant-operator/templates/operator_rbac.yaml]: To be removed

File: charts/tenant-operator/templates/validator_rbac.yaml [hunk 4]
```diff
@@ -95,7 +95,7 @@ metadata:
 subjects:
   - kind: ServiceAccount
     name: {{ include "tenant-validator.hpePrefix" . }}
-    namespace: hpe-system
+    namespace: {{ include "tenant-operator.namespace" . }}
```
Comment (AlexNavara) [inline charts/tenant-operator/templates/validator_rbac.yaml]: Replace with '.Release.Namespace'

**Review by AlexNavara — CHANGES_REQUESTED** [review]

File: charts/tenant-operator/templates/_helpers.tpl (hunk 2 above)
Comment (apurwaj2) [reply charts/tenant-operator/templates/_helpers.tpl]: Removed it, thank you.

**Review by AlexNavara — APPROVED** [review]
lgtm

**Pull request merged**